autoload/suda.vim

```diff
@@ -93,6 +93,31 @@ function! suda#read(expr, ...) abort range
   endtry
 endfunction
 
+function! suda#write_wrapper(action)
+  try 
+    execute('w' . a:action)
+  catch /^Vim\%((\a\+)\)\=:E45/
+    execute('w' . a:action . ' suda://%')
+  endtry
+endfunction
```

Comment:
may

I feel `action` is a bit complicated because it must be trailing text for existing commands.

How about using the following more generic function instead

```vim
function! suda#execute_with_fallback(expr, fallback) abort
  try
    execute(a:expr)
  catch /^Vim\%((\a\+)\)\=:E45:/
    execute(a:fallback)
  endtry
endfunction

" ---

command! W suda#execute_with_fallback('w %', 'w suda://%')
```

Reply:
I've updated it to something a little different I hope you'll like as well, along with only `w` and `wq` abbreviations - no `wa` or `wqa` since I don't we think we should handle that as well.